File: dis-ingest/src/main/java/io/pivotal/dis/ingest/service/store/AmazonS3FileStore.java
package io.pivotal.dis.ingest.service.store;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.model.AccessControlList;
import com.amazonaws.services.s3.model.GroupGrantee;
import com.amazonaws.services.s3.model.Permission;
import com.amazonaws.services.s3.model.PutObjectRequest;
import org.apache.commons.io.IOUtils;

public class AmazonS3FileStore implements FileStore {

    private final AmazonS3 amazonS3;
    private final String bucketName;
    private final AccessControlList disruptionsAcl;

    public AmazonS3FileStore(AmazonS3 amazonS3, String bucketName, AccessControlList accessControlList) {
        this.amazonS3 = amazonS3;
        this.bucketName = bucketName;
        disruptionsAcl = accessControlList;
    }

    @Override
    public void save(String name, String input) {
        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, name, IOUtils.toInputStream(input), null).withAccessControlList(disruptionsAcl);
        amazonS3.putObject(putObjectRequest);
    }
}

File: dis-ingest/src/main/java/io/pivotal/dis/ingest/store/AmazonS3FileStore.kt
package io.pivotal.dis.ingest.store

import com.amazonaws.services.s3.AmazonS3
import com.amazonaws.services.s3.model.AccessControlList
import com.amazonaws.services.s3.model.GroupGrantee
import com.amazonaws.services.s3.model.Permission
import com.amazonaws.services.s3.model.PutObjectRequest
import org.apache.commons.io.IOUtils

class AmazonS3FileStore(private val amazonS3: AmazonS3,
                        private val bucketName: String,
                        private val disruptionsAcl: AccessControlList) : FileStore {

    override fun save(name: String, input: String) {
        val putObjectRequest = PutObjectRequest(
                bucketName,
                name,
                IOUtils.toInputStream(input),
                null).withAccessControlList(disruptionsAcl)

        amazonS3.putObject(putObjectRequest)
    }
}
